feat(auth): use WebView for authentication

machour opened the pull request:
Closes #109 and #221 

Rather than using another library to handle Android, I went ahead and used the WebView component from ReactNative. This seems to be working fine for me on iOS Simulator and my Android Device.

I would really love to have a thorough review of my code, as I feel I've kinda abused the view system here. (and it's the app entry point, so it's a *really* critical change 😆 )

machour commented:
Thanks! Although it still seems buggy on my device, I sometime find myself logged in with an empty events tab and nothing clickable. So still WIP.

housseindjirdeh commented on src/auth/screens/login.screen.js:
```diff
@@ -39,6 +47,32 @@ const styles = StyleSheet.create({
+    height: '100%',
+    width: '100%',
+    textAlign: 'center',
+    backgroundColor: '#1f2327',
```
Let's put `#1f2327` in the `colors` file. Can call it whatever you think sounds reasonable (`githubDark` or something if you like).

machour commented:
@housseindjirdeh I think I fixed the problem with iOS by using Linking only for Android.
I also disabled the cancel button when needed, revamped the Welcome screen to be consistant and cleaned up my UI mess.
Could you give it a new try?

lex111 commented on src/auth/screens/login.screen.js:
```diff
@@ -183,6 +184,22 @@ class Login extends Component {
+  onLoadEnd = e => {
+    const url = e.nativeEvent.url;
+
+    if (url === 'https://github.com/session') {
```
I'm just thinking out loud: can made a separate method, because this code line two times is used, then the meaning of this check will be more clear?

machour replied:
Great advice, thanks! Refactored here: https://github.com/gitpoint/git-point/pull/323/commits/c2c4d565b591eae4b50b05a3ec2ba3b0041d0213

machour commented:
I went ahead and fixed #182 in this same PR by resetting the WebView's cookie as per @andrewda recommendation. The logout is now instantaneous (maybe even too quick 💭 )

I used https://github.com/joeferraro/react-native-cookies to clear the cookies, which needs some extra step in XCode and Android build, documented in their README. I didn't commit the `ios/` and `android/`directories cause I had to tweak a lot of weird things to have a working environment in the same place. I'd rather have one of you guys with a cleaner environment do it when merging this PR.

andrewda commented on src/locale/languages/en.js:
```diff
@@ -4,6 +4,9 @@
   auth: {
     login: {
+      connectingToGitHub: 'Connecting to GitHub..',
+      preparingGitPoint: 'Preparing GitPoint..',
```
Add an extra period to these: `Connecting to GitHub...` and `Preparing GitPoint...`.

lex111 replied:
@machour and more you need to add new keys to other languages, otherwise there will be errors.

machour replied:
@andrewda thanks, fixed!
@lex111 currently if a string is missing, it's displayed with an error `Missing translation for tr.auth.login.cancel`. Isn't that good enough while working on master? Or should I introduce the keys in all language with the english sentence?

lex111 replied:
@machour yes, you need to add English sentences to all languages, just so that there are no such errors.

machour replied:
@lex111 done buddy! I think I'll tackle down i18n right after auth & markdown 😄

lex111 commented on src/locale/languages/fr.js:
```diff
@@ -2,6 +2,9 @@
   auth: {
     login: {
+      connectingToGitHub: 'Connexion à GitHub..',
+      preparingGitPoint: 'Configuration de GitPoint..',
```
Missing dot :smile:

machour replied:
not in french ;)

lex111 replied:
Wow, I did not know :smiley:

lex111 commented on src/auth/screens/login.screen.js:
```diff
@@ -236,6 +236,7 @@ class Login extends Component {
           <View style={styles.container}>
             <Modal
               animationType="slide"
+              onRequestClose={() => null}
```
:+1:

machour commented on ios/GitPoint.xcodeproj/project.pbxproj:
```diff
@@ -623,6 +642,7 @@
 			isa = PBXGroup;
 			children = (
 				273F19201F1E1AD5004B4410 /* libCodePush.a */,
+				48F9DE9B1F6DB8C60064194C /* libCodePush.a */,
```
@housseindjirdeh not sure why Xcode added this line, you may want to have a closer look

machour commented on ios/GitPoint.xcodeproj/project.pbxproj:
```diff
@@ -1148,6 +1181,20 @@
 			remoteRef = 48F49F661F19026D0012FAD6 /* PBXContainerItemProxy */;
 			sourceTree = BUILT_PRODUCTS_DIR;
 		};
+		48F9DE9B1F6DB8C60064194C /* libCodePush.a */ = {
```
same here @housseindjirdeh

machour commented:
@housseindjirdeh @andrewda @lex111 
I just cleaned my project and was able to include links in `android/` and `ios/` for `react-native-cookies`. I thing this PR is all set now.

I'd love to see this merged soon in order for it to be fully tested by everyone before making it to the next release 🙏
